e2e/terraform/provision-infra/userdata/windows-2022.ps1

```diff
@@ -0,0 +1,180 @@
+    throw
+}
+
+Write-Output "Installed Consul."
+
+# -------------------------------------------
+# Install service and firewall rules for Nomad
+# Note the service can't run until we upload Nomad too
+
+Try {
+    New-NetFirewallRule `
+      -DisplayName 'Nomad HTTP Inbound' `
+      -Profile @('Public', 'Domain', 'Private') `
+      -Direction Inbound `
+      -Action Allow `
+      -Protocol TCP `
+      -LocalPort @('4646')
+
+    New-Service `
+      -Name "Nomad" `
+      -BinaryPathName "C:\opt\nomad.exe agent -config C:\etc\nomad.d" `
+      -StartupType "Automatic" `
+      -ErrorAction Ignore
+} Catch {
+    Write-Output "Failed to install Nomad."
+    Write-Output $_
+    $host.SetShouldExit(-1)
+    throw
+}
+
+Write-Output "Installed Nomad."
+
+# --------------------------------------------
+# Install firewall rules required to allow tests
+
+Try {
+    New-NetFirewallRule `
+      -DisplayName 'Metrics Inbound' `
+      -Profile @('Public', 'Domain', 'Private') `
+      -Direction Inbound `
+      -Action Allow `
+      -Protocol TCP `
+      -LocalPort @('6120')
+} Catch {
+    Write-Output "Failed to install firewall rules."
+    Write-Output $_
+    $host.SetShouldExit(-1)
+    throw
+}
+
+# -------------------------------------------
+# Install and configure ssh
+
+# Note: we don't set powershell as the default ssh shell because of
+# https://github.com/hashicorp/terraform/issues/30661
+
+# Note: this is after we install services and binaries so that we can block on
+# ssh availability and not race with the provisioning steps in Terraform
+
+Write-Host 'Installing and starting sshd'
+Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
+Set-Service -Name sshd -StartupType Automatic
+Start-Service sshd
+
+Write-Host 'Installing and starting ssh-agent'
+Add-WindowsCapability -Online -Name OpenSSH.Client~~~~0.0.1.0
+Set-Service -Name ssh-agent -StartupType Automatic
+Start-Service ssh-agent
+
+# From https://learn.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse?tabs=powershell&pivots=windows-server-2022
+# Confirm the Firewall rule is configured. It should be created automatically by
+# setup. Run the following to verify
+if (!(Get-NetFirewallRule -Name "OpenSSH-Server-In-TCP" -ErrorAction SilentlyContinue | Select-Object Name, Enabled)) {
+    Write-Output "Firewall Rule 'OpenSSH-Server-In-TCP' does not exist, creating it..."
+    New-NetFirewallRule -Name 'OpenSSH-Server-In-TCP' -DisplayName 'OpenSSH Server (sshd)' -Enabled True -Direction Inbound -Protocol TCP -Action Allow -LocalPort 22
+} else {
+    Write-Output "Firewall rule 'OpenSSH-Server-In-TCP' has been created and exists."
+}
+
+md "C:\Users\Administrator\.ssh\"
+
+$myKey = "C:\Users\Administrator\.ssh\authorized_keys"
+$adminKey = "C:\ProgramData\ssh\administrators_authorized_keys"
+
+Invoke-RestMethod `
+  -Uri "http://169.254.169.254/latest/meta-data/public-keys/0/openssh-key" `
+  -Outfile $myKey
+
+cp $myKey $adminKey
```

Comment:
this means that the pubkey is set up outside the instance, aws-side, then we grab that and put it in our authorized_keys because we set up ssh manually, yes?